Challenge1/src/agent.spec.ts

```diff
@@ -1,74 +1,127 @@
-//       );
-//       expect(findings).toStrictEqual([
-//         Finding.fromObject({
-//           name: "High Tether Transfer",
-//           description: `High amount of USDT transferred: ${normalizedValue}`,
-//           alertId: "FORTA-1",
-//           severity: FindingSeverity.Low,
-//           type: FindingType.Info,
-//           metadata: {
-//             to: mockTetherTransferEvent.args.to,
-//             from: mockTetherTransferEvent.args.from,
-//           },
-//         }),
-//       ]);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-//         ERC20_TRANSFER_EVENT,
-//         TETHER_ADDRESS
-//       );
-//     });
-//   });
-// });
+  it("returns empty findings if not to Forta Registry", async () => {
+    mockTxEvent.setFrom(mockNethermindAddress).setTo(mockOtherAddress).setData(encodedCreateAgentData);
+    const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns empty findings if neither createAgent nor updateAgent are detected", async () => {
+    mockTxEvent.setFrom(mockNethermindAddress).setTo(mockFortaRegistryAddress).setData(encodedOtherFunctionData);
+    const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns empty findings if transaction calls createAgent but in wrong contract", async () => {
+    mockTxEvent.setFrom(mockNethermindAddress).setTo(mockOtherAddress).setData(encodedCreateAgentData);
+    const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns empty findings if transaction calls updateAgent but in wrong contract", async () => {
+    mockTxEvent.setFrom(mockNethermindAddress).setTo(mockOtherAddress).setData(encodedUpdateAgentData);
+    const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("detects bot deployment", async () => {
+    mockTxEvent.setFrom(mockNethermindAddress).setTo(mockFortaRegistryAddress).setData(encodedCreateAgentData);
+
+    // .addTraces({
+    //   from: mockNethermindAddress,
+    //   to: mockFortaRegistryAddress,
+    //   function: CREATE_AGENT_ABI,
+    //   arguments: [MOCK_AGENT_ID, mockNethermindAddress, "metadata", CHAIN_IDS],
+    // });
+    // .setBlock(60343606);
+    // .setBlock(56681086);
+
+    const findings = await handleTransaction(mockTxEvent);
+    findings.push(...(await handleTransaction(mockTxEvent2)));
+    expect(findings).toHaveLength(1);
+    expect(findings).toStrictEqual([
+      Finding.fromObject({
+        name: "Nethermind Bot Deployment",
+        description: "Nethermind deployed a new bot with ID: 1",
+        alertId: "NEW-BOT-DEPLOYED",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+        metadata: {
+          agentId: "1",
+          chainId: "137",
+        },
+      }),
+    ]);
```

Comment:
I understand the thinking behind this, _but_ this isn't actually simulating how the bot would actually work in production. In production, the bot would process only one transaction at a time and return findings found in that singular transaction.

In this test, and the following one, the `findings` array is actually findings from two separate transactions. So it's not mocking what actually happens in production.

The correct way would be to include the logs in the same _single_ transaction, and the bot would filter out the one that doesn't meet the criteria.

if you need an example of adding multiple logs in a single transaction, take a look at this [here](https://github.com/NethermindEth/Forta-Agents/blob/main/DYDX-Bots/Large-USDC-Staking-Withdrawals/src/agent.spec.ts#L266-L272). However, please note that that implementation uses _anonymous_ logs, but the `TestTransactionEvent` class actually also has an `.addEventLog()` method. Docs are [here](https://github.com/NethermindEth/general-agents-module?tab=readme-ov-file#testtransactionevent).